integration/src/main/java/scripts/api/TdrDatasetsApi.java

```diff
@@ -0,0 +1,166 @@
+package scripts.api;
+
+import bio.terra.datarepo.api.ProfilesApi;
+import bio.terra.datarepo.client.ApiException;
+import bio.terra.datarepo.model.ColumnModel;
+import bio.terra.datarepo.model.DatasetModel;
+import bio.terra.datarepo.model.DatasetRequestModel;
+import bio.terra.datarepo.model.DatasetSpecificationModel;
+import bio.terra.datarepo.model.IngestRequestModel;
+import bio.terra.datarepo.model.PolicyMemberRequest;
+import bio.terra.datarepo.model.TableDataType;
+import bio.terra.datarepo.model.TableModel;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import scripts.client.DatarepoClient;
+
+public class TdrDatasetsApi {
+  private static final Logger log = LoggerFactory.getLogger(TdrDatasetsApi.class);
+  private static final String TEST_BILLING_PROFILE_PREFIX = "catalog_test_profile";
+  private static TdrDatasetsApi theApi;
+  private DatasetModel testDataset;
```

Comment:
I think making the class a singleton makes it easer to reason about how it will work, but that's a pretty subjective measure. In this case though, `datasetsApi` must also be static, since it's needed to delete the dataset at shutdown time, so I think it's simpler to make the class a singleton rather than make all its fields static.

I prefer using singleton objects with non-static methods in general (vs static methods with static members), since you can still have "real" static methods that don't rely on any class state.

Reply:
Sounds good. Although I think it's worth leaving a comment explaining why we have a singleton, as without knowledge of the TDR locking issue, it's a puzzling choice.